contracts/EverlongStrategyFactory.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { IEverlongStrategyFactory } from "./interfaces/IEverlongStrategyFactory.sol";
+import { IEverlongStrategy } from "./interfaces/IEverlongStrategy.sol";
+import { EVERLONG_STRATEGY_FACTORY_KIND, EVERLONG_VERSION } from "./libraries/Constants.sol";
+import { EverlongStrategy } from "./EverlongStrategy.sol";
+
+// TODO: Add testing.
+//
+/// @author DELV
+/// @title EverlongStrategyFactory
+/// @notice A factory for creating EverlongStrategy instances.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract EverlongStrategyFactory is IEverlongStrategyFactory {
+    /// @inheritdoc IEverlongStrategyFactory
+    string public name;
+
+    /// @inheritdoc IEverlongStrategyFactory
+    string public constant kind = EVERLONG_STRATEGY_FACTORY_KIND;
+
+    /// @inheritdoc IEverlongStrategyFactory
+    string public constant version = EVERLONG_VERSION;
+
+    /// @inheritdoc IEverlongStrategyFactory
+    address public immutable emergencyAdmin;
```

Comment:
What's the rationale for this being immutable?

Reply:
The bulk of this contract was taken from https://github.com/yearn/tokenized-strategy-foundry-mix/blob/master/src/StrategyFactory.sol and that's how they had it so I left it. Personally don't expect us to be changing these values often so this seemed fine to me but open to different perspectives

Reply:
Gotcha. It seems like it would be useful to be able to rotate the address. Having it be mutable would be my preference.